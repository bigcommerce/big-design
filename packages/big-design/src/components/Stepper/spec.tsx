import 'jest-styled-components';
import { render, screen } from '@testing-library/react';
import React from 'react';

import { Stepper } from './';

const steps: string[] = ['Login', 'Settings', 'Import'];

test('renders Stepper', () => {
  render(<Stepper currentStep={1} steps={steps} />);

  expect(screen.getByRole('progressbar')).toMatchSnapshot();
});

test('does not forward styles', () => {
  render(
    <Stepper className="test" currentStep={0} steps={steps} style={{ backgroundColor: 'red' }} />,
  );

  expect(screen.queryByRole('progressbar', { name: /test/i })).not.toBeInTheDocument();
  expect(screen.getByRole('progressbar')).not.toHaveStyle('background: red');
});

test('renders a passed step', () => {
  render(<Stepper currentStep={0} steps={['Test']} />);

  expect(screen.getByText('Test')).toBeInTheDocument();
});

test('passed in state works as expected', () => {
  render(<Stepper currentStep={1} steps={['Test1', 'Test2', 'Test3']} />);

  const lights: ChildNode[] = [];
  const dashes: ChildNode[] = [];

  screen.queryByRole('progressbar')?.childNodes.forEach((step) => {
    lights.push(step.childNodes[0]);
    dashes.push(step.childNodes[1]);
  });

  // eslint-disable-next-line testing-library/no-node-access
  expect(lights[0].firstChild).not.toHaveTextContent('1');
  // eslint-disable-next-line testing-library/no-node-access
  expect(lights[1].firstChild).toHaveTextContent('2');
  // eslint-disable-next-line testing-library/no-node-access
  expect(lights[2].firstChild).toHaveTextContent('3');

  expect(dashes[0]).toHaveStyle('background-color: #3C64F4');
  expect(dashes[1]).toHaveStyle('background-color: #D9DCE9');
  expect(dashes[2]).toHaveStyleRule('display', 'none', {
    media: '(min-width:0px)',
  });
});
