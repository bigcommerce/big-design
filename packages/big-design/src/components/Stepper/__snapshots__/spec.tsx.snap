// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`renders Stepper 1`] = `
.c0 {
  margin-bottom: 1.5rem;
  box-sizing: border-box;
}

.c2 {
  margin-bottom: 1.5rem;
  box-sizing: border-box;
}

.c5 {
  box-sizing: border-box;
  background-color: #3C64F4;
}

.c9 {
  box-sizing: border-box;
  background-color: #3C64F4;
}

.c13 {
  margin-bottom: 1.5rem;
  box-sizing: border-box;
}

.c16 {
  box-sizing: border-box;
  background-color: #D9DCE9;
}

.c17 {
  box-sizing: border-box;
  background-color: #8C93AD;
}

.c18 {
  box-sizing: border-box;
  background-color: #D9DCE9;
}

.c1 {
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c6 {
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c8 {
  vertical-align: middle;
  color: #FFFFFF;
  height: 1.25rem;
  width: 1.25rem;
}

.c3 {
  gap: 1rem;
  display: grid;
}

.c14 {
  gap: 1rem;
  display: grid;
}

.c11 {
  color: #313440;
  margin: 0 0 1rem;
  font-size: 1rem;
  font-weight: 400;
  line-height: 1.5rem;
}

.c11:last-child {
  margin-bottom: 0;
}

.c19 {
  color: #8C93AD;
  margin: 0 0 1rem;
  font-size: 1rem;
  font-weight: 400;
  line-height: 1.5rem;
}

.c19:last-child {
  margin-bottom: 0;
}

.c4 {
  -webkit-flex-basis: 100%;
  -ms-flex-preferred-size: 100%;
  flex-basis: 100%;
  gap: 0.5rem;
  grid-template-areas: 'light dash' 'text text';
  grid-template-columns: auto 1fr;
}

.c7 {
  border-radius: 1rem;
  color: #FFFFFF;
  grid-area: light;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c10 {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  height: 0.125rem;
  grid-area: dash;
  width: 100%;
}

.c12 {
  grid-area: text;
}

@media (min-width:0px) {
  .c2 {
    display: none;
  }
}

@media (min-width:720px) {
  .c2 {
    display: grid;
  }
}

@media (min-width:720px) {
  .c2 {
    margin-right: 0.5rem;
  }
}

@media (min-width:0px) {
  .c5 {
    padding-top: 0.25rem;
    padding-bottom: 0.25rem;
  }
}

@media (min-width:720px) {
  .c5 {
    padding-top: 0;
    padding-bottom: 0;
  }
}

@media (min-width:0px) {
  .c5 {
    padding-left: 0.5rem;
    padding-right: 0.5rem;
  }
}

@media (min-width:720px) {
  .c5 {
    padding-left: 0;
    padding-right: 0;
  }
}

@media (min-width:0px) {
  .c9 {
    display: none;
  }
}

@media (min-width:720px) {
  .c9 {
    display: block;
  }
}

@media (min-width:0px) {
  .c13 {
    display: grid;
  }
}

@media (min-width:720px) {
  .c13 {
    display: grid;
  }
}

@media (min-width:720px) {
  .c13 {
    margin-right: 0.5rem;
  }
}

@media (min-width:0px) {
  .c16 {
    display: none;
  }
}

@media (min-width:720px) {
  .c16 {
    display: block;
  }
}

@media (min-width:0px) {
  .c17 {
    padding-top: 0.25rem;
    padding-bottom: 0.25rem;
  }
}

@media (min-width:720px) {
  .c17 {
    padding-top: 0;
    padding-bottom: 0;
  }
}

@media (min-width:0px) {
  .c17 {
    padding-left: 0.5rem;
    padding-right: 0.5rem;
  }
}

@media (min-width:720px) {
  .c17 {
    padding-left: 0;
    padding-right: 0;
  }
}

@media (min-width:0px) {
  .c18 {
    display: none;
  }
}

@media (min-width:720px) {
  .c18 {
    display: none;
  }
}

@media (min-width:0px) {
  .c3 {
    display: none;
  }
}

@media (min-width:720px) {
  .c3 {
    display: grid;
  }
}

@media (min-width:0px) {
  .c14 {
    display: grid;
  }
}

@media (min-width:720px) {
  .c14 {
    display: grid;
  }
}

@media (min-width:720px) {
  .c4 {
    grid-template-columns: 1.5rem 1fr;
    grid-template-rows: 1.5rem 1fr;
    max-width: 14.25rem;
    min-width: 6rem;
  }
}

@media (min-width:720px) {
  .c15 {
    border: 0;
    -webkit-clip: rect(0 0 0 0);
    clip: rect(0 0 0 0);
    height: 1px;
    margin: -1px;
    overflow: hidden;
    padding: 0;
    position: absolute;
    white-space: nowrap;
    width: 1px;
  }
}

@media (min-width:720px) {
  .c7 {
    border-radius: 50%;
    font-size: 1rem;
    line-height: 1.5rem;
  }
}

<div
  aria-valuemax="3"
  aria-valuemin="0"
  aria-valuenow="2"
  aria-valuetext="Step 2: Settings"
  class="c0 c1"
  role="progressbar"
>
  <div
    class="c2 c3 c4"
    display="[object Object]"
  >
    <div
      class="c5 c6 c7"
    >
      <svg
        aria-hidden="true"
        class="c8"
        color="white"
        fill="currentColor"
        height="24"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="24"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M9 16.17 5.53 12.7a.996.996 0 1 0-1.41 1.41l4.18 4.18c.39.39 1.02.39 1.41 0L20.29 7.71a.996.996 0 1 0-1.41-1.41z"
        />
      </svg>
    </div>
    <div
      class="c9 c10"
      display="[object Object]"
    />
    <p
      class="c11 c12"
      color="secondary70"
    >
      Login
    </p>
  </div>
  <div
    class="c13 c14 c4"
    display="[object Object]"
  >
    <div
      class="c5 c6 c7"
    >
      <span>
        <span
          class="c15"
        >
          Step
        </span>
         2 
        <span
          class="c15"
        >
          of 
          3
        </span>
      </span>
    </div>
    <div
      class="c16 c10"
      display="[object Object]"
    />
    <p
      class="c11 c12"
      color="secondary70"
    >
      Settings
    </p>
  </div>
  <div
    class="c2 c3 c4"
    display="[object Object]"
  >
    <div
      class="c17 c6 c7"
    >
      <span>
        <span
          class="c15"
        >
          Step
        </span>
         3 
        <span
          class="c15"
        >
          of 
          3
        </span>
      </span>
    </div>
    <div
      class="c18 c10"
      display="[object Object]"
    />
    <p
      class="c19 c12"
      color="secondary50"
    >
      Import
    </p>
  </div>
</div>
`;
